lib/ansible_tower_client/middleware/raise_tower_error.rb

```diff
@@ -13,6 +14,10 @@ def on_complete(env)
           # mimic the behavior that we get with proxy requests with HTTPS
           raise AnsibleTowerClient::ConnectionError, %(407 "Proxy Authentication Required ")
         when CLIENT_ERROR_STATUSES
+          logger.debug { "#{self.class.name} Raw Response:\n#{env.pretty_inspect}" }
+          message   = JSON.parse(env.body).pretty_inspect rescue nil
+          message ||= env.body
+          logger.error("#{self.class.name} Response Body:\n#{message}")
```

Comment:
@bdunne - I like putting this here.  

It would probably be good to get some specs around the logging in Middleware.